File: shoring/blackjackEngine/src/test/java/com/panopset/tests/blackjackEngine/ShoeTest.java
package com.panopset.tests.blackjackEngine;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;
import org.junit.jupiter.api.Test;

import com.panopset.blackjackEngine.Card;
import com.panopset.blackjackEngine.DeckPile;
import com.panopset.blackjackEngine.Shoe;

public class ShoeTest {

  @Test
  void testShuffle() {
    DeckPile.reset();
    Shoe shoe = new Shoe(5, 1);
    assertFalse(shoe.isTheDeckStacked());
    assertEquals(5, shoe.getNumberOfDecks());
    Card card = shoe.deal();
    assertTrue(card.isBlue());
    assertNotNull(card);
    assertTrue(shoe.isBlue());
    assertTrue(shoe.deal().isBlue());
    shoe.shuffle();
    assertFalse(shoe.isBlue());
    assertFalse(shoe.deal().isBlue());
  }
  
  @Test
  void testDeal() {
    DeckPile.reset();
    Shoe shoe = new Shoe(1, 1);
    assertFalse(shoe.isTheDeckStacked());
    assertTrue(shoe.isBlue());
    shoe.deal();
    for (int i = 0; i < 53; i++) {
      shoe.deal();
    }
    assertFalse(shoe.isBlue());
  }
}

File: code/shoring/blackjackEngine/src/test/kotlin/com/panopset/tests/blackjackEngine/ShoeTest.kt
package com.panopset.tests.blackjackEngine

import com.panopset.blackjackEngine.DeckPile.reset
import com.panopset.blackjackEngine.Shoe
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class ShoeTest {
    @Test
    fun testShuffle() {
        reset()
        val shoe = Shoe(5)
        Assertions.assertFalse(shoe.isTheDeckStacked())
        Assertions.assertEquals(5, shoe.numberOfDecks)
        val card = shoe.deal()
        Assertions.assertTrue(card.isBlue)
        Assertions.assertNotNull(card)
        Assertions.assertTrue(shoe.isBlue)
        Assertions.assertTrue(shoe.deal().isBlue)
        shoe.shuffle()
        Assertions.assertFalse(shoe.isBlue)
        Assertions.assertFalse(shoe.deal().isBlue)
    }

    @Test
    fun testDeal() {
        reset()
        val shoe = Shoe(1)
        Assertions.assertFalse(shoe.isTheDeckStacked())
        Assertions.assertTrue(shoe.isBlue)
        shoe.deal()
        for (i in 0..52) {
            shoe.deal()
        }
        Assertions.assertFalse(shoe.isBlue)
    }
}
